lib/double_entry/reporting/line_metadata_filter.rb

```diff
@@ -0,0 +1,33 @@
+# encoding: utf-8
+module DoubleEntry
+  module Reporting
+    class LineMetadataFilter
+
+      def self.filter(collection:, metadata:)
+        table_alias_index = 0
+
+        metadata.reduce(collection) do |filtered_collection, (key, value)|
+          table_alias = "m#{table_alias_index}"
+          table_alias_index += 1
+
+          filtered_collection.
+            joins("INNER JOIN #{line_metadata_table} as #{table_alias} ON #{table_alias}.line_id = #{lines_table}.id").
```

Comment:
Yup - So the first ends up using a filesort, the second doesn't.

I think this is a case of: if we find ourselves needing to add more and more metadata, we need to revisit this.  For our case we just need to filter by two, so we should be okay with the memory used by the joins.